frb_codegen/src/commands.rs

```diff
@@ -274,3 +283,88 @@ pub fn build_runner(dart_root: &str) -> Result {
     }
     Ok(())
 }
+
+pub fn guess_context(dart_root: &str) -> std::result::Result<Context, Error> {
+    info!("Guessing context the runner is run into");
+    let pubspec = PathBuf::from(dart_root).join("pubspec.lock");
+    if !pubspec.exists() {
+        return Err(Error::StringError(format!(
+            "missing pubspec.lock in {}",
+            dart_root
+        )));
+    }
+    let pubspec = std::fs::read_to_string(pubspec)
+        .map_err(|_| Error::StringError(format!("unable to read pubspec.lock in {}", dart_root)))?;
+    let pubspec: Pubspec = serde_yaml::from_str(&pubspec).map_err(|_| {
+        Error::StringError(format!("unable to parse pubspec.lock in {}", dart_root))
+    })?;
+    if pubspec.packages.contains_key("flutter") {
+        return Ok(Context::Flutter);
+    }
+    Ok(Context::Dart)
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Context {
+    Dart,
+    Flutter,
+}
+
+#[derive(Debug, Deserialize)]
+struct Pubspec {
+    pub packages: HashMap<String, Info>,
+}
+
+#[derive(Debug, Deserialize)]
+#[allow(dead_code)]
```

Comment:
uber-nit: can this be even an empty struct? s.t. we do not need to add `allow(dead_code)`

Reply:
oh well spotted indeed 👌 probably like 8ab66e0bfcecb08475e2ee903e665663407937aa